src/lib.rs

```diff
@@ -270,8 +270,20 @@ fn open_browser_internal(browser: Browser, url: &str) -> Result<ExitStatus> {
     match browser {
         Browser::Default => open_on_unix_using_browser_env(url)
             .or_else(|_| -> Result<ExitStatus> { Command::new("xdg-open").arg(url).status() })
+            .or_else(|r| -> Result<ExitStatus> {
+                if let Ok(desktop) = ::std::env::var("XDG_CURRENT_DESKTOP") {
+                    if desktop == "KDE" {
+                        return Command::new("kioclient").arg("exec").arg(url).status();
+                    }
+                }
+                Err(r) // If either `if` check fails, fall through to the next or_else
+            })
             .or_else(|_| -> Result<ExitStatus> { Command::new("gvfs-open").arg(url).status() })
-            .or_else(|_| -> Result<ExitStatus> { Command::new("gnome-open").arg(url).status() }),
+            .or_else(|_| -> Result<ExitStatus> { Command::new("gnome-open").arg(url).status() })
+            .or_else(|_| -> Result<ExitStatus> {
+                Command::new("kioclient").arg("exec").arg(url).status()
```

Comment:
Do we need this when we've already checked for `KDE` via `XDG_CURRENT_DESKTOP`?

Reply:
Yes. Its purpose is to ensure that `kioclient exec` still gets tried *after* `gvfs-open` and `gnome-open` when `XDG_CURRENT_DESKTOP` does not equal `KDE`.

For example, suppose the user is running one of these environments:

* KDE applications under an Openbox session
* LXQt, which borrows components from KDE Frameworks 5 but is its own desktop environment.